docs/proposals/apb-state-support.md

```diff
@@ -0,0 +1,59 @@
+# Abstract
+
+This proposal will outline a simple and secure method for allowing Service Bundle developers to set state during a Service Bundle
+action and then be able to access that state during a subsequent Service Bundle action.
+
+## Actions
+   - provision
+   - deprovision
+   - update
+   - bind 
+   - unbind
+   
+   
+## Problem Description
+
+Service Bundle's are stateless. All state is managed for them by the broker. Currently the broker passes in parameters specified
+during a request to the catalog. It also passes in some additional parameters such as the namespace etc, additionally 
+we also pass in credentials created during the provision. While this is useful, there is no mechanism for a Service Bundle to store 
+and access data across actions without exposing it to the end user. We want to avoid Service Bundle developers working around this 
+limitation by doing suboptimal things such as storing extra data in the credentials or by creating a ConfigMap in the user's 
+namespace as both of these expose the data to the user and, as the namespace is controlled by the user, it is naturally 
+not trustworthy or reliable.
+
+
+
+# Proposed Solution
+
+## Service Bundle contract for handling state
+
+### Add APB specific module
+
+This module will expose an API to the APB developer for setting key value pairs.
+
+```
+- name: Save some stuff
+  asb_set_state:
+    service_name: "{{ service_name }}"
```

Comment:
`service_name` in this case is the arbitrary state data, correct? Just want to make sure it's not suggested that this is some kind of special key.

Reply:
just arbitrary